esphome/components/samsung/samsung.cpp

```diff
@@ -0,0 +1,238 @@
+#include "samsung.h"
+
+namespace esphome {
+namespace samsung {
+
+void SamsungClimate::transmit_state() {
+  if (current_climate_mode_ != climate::ClimateMode::CLIMATE_MODE_OFF &&
+      this->mode == climate::ClimateMode::CLIMATE_MODE_OFF) {
+    set_and_send_power_state_(false);
+    return;
+  }
+
+  if (current_climate_mode_ == climate::ClimateMode::CLIMATE_MODE_OFF &&
+      this->mode != climate::ClimateMode::CLIMATE_MODE_OFF) {
+    set_and_send_power_state_(true);
+  }
+
+  current_climate_mode_ = this->mode;
+
+  set_mode_(this->mode);
+  set_temp_(this->target_temperature);
+  set_swing_(this->swing_mode);
+  set_fan_(this->fan_mode.has_value() ? this->fan_mode.value() : climate::CLIMATE_FAN_AUTO);
+
+  send_();
+}
+
+/// Send the current state of the climate object.
+void SamsungClimate::send_() {
+  checksum_();
+
+  auto transmit = this->transmitter_->transmit();
+  auto *data = transmit.get_data();
+  data->set_carrier_frequency(SAMSUNG_IR_FREQUENCY);
+
+  // Header
+  data->mark(SAMSUNG_AIRCON1_HDR_MARK);
+  data->space(SAMSUNG_AIRCON1_HDR_SPACE);
+
+  for (int i = 0; i < 21; i++) {
+    if (i == 7 || i == 14) {
+      data->mark(SAMSUNG_AIRCON1_BIT_MARK);
+      data->space(SAMSUNG_AIRCON1_MSG_SPACE);
+
+      data->mark(SAMSUNG_AIRCON1_HDR_MARK);
+      data->space(SAMSUNG_AIRCON1_HDR_SPACE);
+    }
+
+    uint8_t send_byte = protocol_.raw[i];
+
+    for (int y = 0; y < 8; y++) {
+      if (send_byte & 0x01) {
+        data->mark(SAMSUNG_AIRCON1_BIT_MARK);
+        data->space(SAMSUNG_AIRCON1_ONE_SPACE);
+      } else {
+        data->mark(SAMSUNG_AIRCON1_BIT_MARK);
+        data->space(SAMSUNG_AIRCON1_ZERO_SPACE);
```

Comment:
```suggestion
        data->item(SAMSUNG_AIRCON1_BIT_MARK, SAMSUNG_AIRCON1_ZERO_SPACE);
```
https://github.com/esphome/esphome/blob/387bde665e3ae5d4b8d5a68299abd6032c4291b8/esphome/components/remote_base/remote_base.h#L29-L32